src/app/core/services/kafka.service.ts

```diff
@@ -156,73 +140,75 @@ export class KafkaService {
   }
 
   createPayloadAndSend(specs, kafkaConnInstance) {
-    let schemaVersions
-    switch (specs.name) {
-      case KAFKA_COMPLETION_LOG:
-        if (this.schemas[specs.name]) {
-          schemaVersions = this.schemas[specs.name]
-          break
-        }
-      default:
-        schemaVersions = this.util
-          .getLatestKafkaSchemaVersions(specs)
-          .catch(error => {
-            console.log(error)
-            return Promise.resolve()
-          })
-        this.schemas[specs.name] = schemaVersions
-    }
-    return Promise.all([schemaVersions]).then(data => {
-      schemaVersions = data[0]
-      const avroKey = AvroSchema.parse(
-        JSON.parse(schemaVersions[0]['schema']),
-        {
-          wrapUnions: true
-        }
-      )
-      const avroVal = AvroSchema.parse(
-        JSON.parse(schemaVersions[1]['schema']),
-        {
-          wrapUnions: true
-        }
-      )
-      const kafkaObject = specs.kafkaObject
-      const bufferKey = avroKey.clone(kafkaObject.key, { wrapUnions: true })
-      const bufferVal = avroVal.clone(kafkaObject.value, { wrapUnions: true })
-      const payload = {
-        key: bufferKey,
-        value: bufferVal
+    return this.util.getKafkaTopic(specs).then(topic => {
```

Comment:
What happens when the `getKafkaTopic()` fails?

Reply:
Yup, it would use the `defaultTopic` here: https://github.com/RADAR-base/RADAR-Questionnaire/pull/420/files#diff-fea08f2807ce3af1e06f6e1ce8715184R79